spokestack/nlu/tflite.py

```diff
@@ -0,0 +1,119 @@
+"""
+This module contains the class to serve TFLite NLU models
+"""
+import json
+import os
+from importlib import import_module
+from typing import Any, Dict, List, Tuple
+
+import numpy as np  # type: ignore
+from tokenizers import BertWordPieceTokenizer  # type: ignore
+
+from spokestack import utils
+from spokestack.models.tensorflow import TFLiteModel
+
+
+class TFLiteNLU:
+    """ Abstraction for using TFLite NLU models
+
+    Args:
+        model_dir (str): path to the model directory containing nlu.tflite,
+                         metadata.json, and vocab.txt
+    """
+
+    def __init__(self, model_dir: str) -> None:
+        self._model = TFLiteModel(model_path=os.path.join(model_dir, "nlu.tflite"))
+        self._metadata = utils.load_json(os.path.join(model_dir, "metadata.json"))
+        self._tokenizer = BertWordPieceTokenizer(os.path.join(model_dir, "vocab.txt"))
+        self._max_length = self._model.input_details[0]["shape"][-1]
+        self._intent_decoder = {
+            i: intent["name"] for i, intent in enumerate(self._metadata["intents"])
+        }
+        self._tag_decoder = {i: tag for i, tag in enumerate(self._metadata["tags"])}
+        self._intent_meta = {
+            intent.pop("name"): intent for intent in self._metadata["intents"]
+        }
+        self._warm_up()
+
+    def __call__(self, utterance: str) -> Dict[str, Any]:
+        """ Forward Pass
```

Comment:
Technically I suppose inference is a forward pass, but isn't this terminology mainly for training? I'd elaborate here a bit. This classifies an utterance, returning the intent predicted by the model, its confidence in that prediction, and a dict of slots recognized in the utterance and parsed according to types specified in the model metadata.